lighthouse-cli/test/smokehouse/run-smoke.js

```diff
@@ -0,0 +1,191 @@
+}, {
+  id: 'ttci',
+  expectations: 'tricky-ttci/expectations.js',
+  config: 'lighthouse-core/config/default-config.js',
+}];
+
+/**
+ * Display smokehouse output from child process
+ * @param {{id: string, process: NodeJS.Process} || {id: string, error: Error & {stdout : NodeJS.WriteStream, stderr: NodeJS.WriteStream}}} result
+ */
+function displaySmokehouseOutput(result) {
+  console.log(`\n${purpleify(result.id)} smoketest results:`);
+  if (result.error) {
+    console.log(result.error.message);
+    process.stdout.write(result.error.stdout);
+    process.stderr.write(result.error.stderr);
+  } else {
+    process.stdout.write(result.process.stdout);
+    process.stderr.write(result.process.stderr);
+  }
+  console.timeEnd(`smoketest-${result.id}`);
+  console.log(`${purpleify(result.id)} smoketest complete. \n`);
+}
+
+/**
+ * Run smokehouse in child processes for selected smoketests
+ * Display output from each as soon as they finish, but resolve function when ALL are complete
+ * @param {Array<SmoketestDfn>} smokes
+ */
+async function runSmokehouse(smokes) {
+  const cmdPromises = [];
+  for (const {id, expectations, config} of smokes) {
+    console.log(`${purpleify(id)} smoketest starting…`);
+    console.time(`smoketest-${id}`);
+    const cmd = [
+      'node lighthouse-cli/test/smokehouse/smokehouse.js',
+      `--config-path=${config}`,
+      `--expectations-path=${expectations}`,
+    ].join(' ');
+
+    // The promise ensures we output immediately, even if the process errors
+    const p = execAsync(cmd, {timeout: 3 * 60 * 1000, encoding: 'utf8'}).then(cp => {
+      const ret = {id: id, process: cp};
+      displaySmokehouseOutput(ret);
+      return ret;
+    }).catch(err => {
+      const ret = {id: id, error: err};
+      displaySmokehouseOutput(ret);
+      return ret;
+    });
+
+    // If the machine is terribly slow, we'll run all smoketests in succession, not parallel
+    if (process.env.APPVEYOR) {
+      await p;
+    }
+    cmdPromises.push(p);
+  }
+
+  return Promise.all(cmdPromises);
+}
+
+/**
+ *
+ * @param {string[]} argv
+ * @return {Map<string, Array<SmoketestDfn>>}
+ */
+function getSmoketestBatches(argv) {
+  let smokes = [];
+  const usage = `    ${log.dim}yarn smoke ${SMOKETESTS.map(t => t.id).join(' ')}${log.reset}\n`;
+
+  if (argv.length === 0) {
+    smokes = SMOKETESTS;
+    console.log('Running ALL smoketests. Equivalent to:');
+    console.log(usage);
+  } else {
+    smokes = SMOKETESTS.filter(test => argv.includes(test.id));
+    console.log(`Running ONLY smoketests for: ${smokes.map(t => t.id).join(' ')}\n`);
+  }
+
+  const unmatchedIds = argv.filter(requestedId => !SMOKETESTS.map(t => t.id).includes(requestedId));
+  if (unmatchedIds.length) {
+    console.log(log.redify(`Smoketests not found for: ${unmatchedIds.join(' ')}`));
+    console.log(usage);
+  }
+
+  // Split into serial batches that will run their tests concurrently
+  const batches = smokes.reduce((map, test) => {
```

Comment:
is this still running the two perf batches in parallel?

Reply:
currently there's a default batch (with like 7 smoketests) and a perf batch (with 2).

when perf batch goes, the 2 smoketests in there do run in parallel.

I was hoping that style was a little more clear with this "batch" framing.